Content.Server/Shuttles/Systems/ArrivalsSystem.cs

```diff
@@ -241,6 +260,50 @@ private void OnPlayerSpawn(PlayerSpawningEvent ev)
         }
     }
 
+    private bool TryTeleportToMapSpawn(EntityUid player, EntityUid stationId, TransformComponent? transform = null)
+    {
+        if (!Resolve(player, ref transform))
+            return false;
+
+        var points = EntityQueryEnumerator<SpawnPointComponent, TransformComponent>();
+        var possiblePositions = new List<EntityCoordinates>();
+        var passengerPositions = new List<EntityCoordinates>();
+
+        // Find a spawnpoint on the same map as the player is already docked with now.
+        while ( points.MoveNext(out var uid, out var spawnPoint, out var xform))
+        {
+            if (spawnPoint.SpawnType == SpawnPointType.LateJoin &&
+                _station.GetOwningStation(uid, xform) == stationId)
+            {
+                // Add to list of possible spawn locations
+                possiblePositions.Add(xform.Coordinates);
+            } else if (possiblePositions.Count == 0 && spawnPoint?.Job?.ID == "Passenger" &&
+                       _station.GetOwningStation(uid, xform) == stationId)
```

Comment:
Newline else-ifs, also if you're adding hardcoded prototype IDs use [ValidatePrototypeId<EntityPrototype>] attribute.